smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -55,17 +60,520 @@
 from smartsim._core.launcher.pbs.pbsLauncher import PBSLauncher
 from smartsim._core.launcher.slurm.slurmLauncher import SlurmLauncher
 from smartsim._core.launcher.stepInfo import StepInfo
-from smartsim._core.utils.helpers import get_ts
+from smartsim._core.utils.helpers import get_ts_ms
 from smartsim._core.utils.serialize import MANIFEST_FILENAME
 from smartsim.error.errors import SmartSimError
+from smartsim.log import get_logger
 from smartsim.status import STATUS_COMPLETED, TERMINAL_STATUSES
 
+# pylint: disable=too-many-lines
 """Telemetry Monitor entrypoint"""
 
 # kill is not catchable
 SIGNALS = [signal.SIGINT, signal.SIGQUIT, signal.SIGTERM, signal.SIGABRT]
 _EventClass = t.Literal["start", "stop", "timestep"]
 _MAX_MANIFEST_LOAD_ATTEMPTS: t.Final[int] = 6
+F_MIN, F_MAX = 1.0, 600.0
+_LOG_FILE_NAME = "logs/telemetrymonitor.out"
+
+logger = get_logger(__name__)
+
+
+class Sink(abc.ABC):
+    """Base class for telemetry output sinks"""
+
+    @abc.abstractmethod
+    async def save(self, *args: t.Any) -> None:
+        """Save the passed data to the underlying sink"""
```

Comment:
I'm a little weirded out that we're just passing in the string to save as a set of **kwargs. What're your thoughts on injecting some kind of IR here even if for now, it effectively does what you're doing here.